api/types/kustomization.go

```diff
@@ -172,6 +172,31 @@ type Kustomization struct {
 	BuildMetadata []string `json:"buildMetadata,omitempty" yaml:"buildMetadata,omitempty"`
 }
 
+const (
+	deprecatedBaseWarningMessage           = "# Warning: bases was deprecated, Please use 'resources' or try `kustomize edit fix`."
+	deprecatedPatchesJson6902Message       = "# Warning: patchesJson6902 is deprecated, Please use 'patches' or try `kustomize edit fix`."
+	deprecatedPatchesStrategicMergeMessage = "# Warning: patchesStrategicMerge is deprecated, Please use 'patches'."
+	deprecatedVarsMessage                  = "# Warning: vars is deprecated, Please use 'replacements'."
```

Comment:
@KnVerey Would you like to take a look a these warning messages before we merge?

Reply:
- Please change the commas after "deprecated" to periods.
- Please quote the names of the old fields, to match the names of the new.
- Since the point of `kustomize edit fix` is to do these updates for you, it isn't really an _alternative_ to using the new fields. How about `Warning: 'X' is deprecated. Please use 'Y' instead. Run 'kustomize edit fix' to update your Kustomization automatically.`.
- For replacements, we can add something like "(EXPERIMENTAL)" to the edit fix part.
- If this merges before https://github.com/kubernetes-sigs/kustomize/pull/4733, that PR will need to update the message.

Reply:
Hi @KnVerey 
I think I completed to fix from your comment.

> If this merges before https://github.com/kubernetes-sigs/kustomize/pull/4733, that PR will need to update the message.

I added a warning message to `patchesStrategicMerge` in this PR.
I think it is better to merge first to https://github.com/kubernetes-sigs/kustomize/pull/4733 if I can.

@natasha41575 
Would you review https://github.com/kubernetes-sigs/kustomize/pull/4733 before merging this PR?